Kinvey/KinveyTests/MultiInsertSpec.swift

```diff
@@ -0,0 +1,1636 @@
+                        ]
+                    ).result
+                    expect(result).toNot(beNil())
+                    guard let resultUnwrapped = result else {
+                        return
+                    }
+                    expect(resultUnwrapped.entities.count).to(equal(2))
+                    expect(resultUnwrapped.errors.count).to(equal(0))
+                
+                    expect(syncDataStore.pendingSyncCount()).to(equal(2))
+                    expect(syncDataStore.pendingSyncEntities().count).to(equal(2))
+                    expect(syncDataStore.pendingSyncOperations().count).to(equal(1))
+                    
+                    let errors = kinveyPush(dataStore: syncDataStore).errors
+                    expect(errors?.count).to(equal(1))
+                    expect(errors?.first?.localizedDescription).to(equal("The Kinvey server encountered an unexpected error. Please retry your request."))
+                    
+                    expect(syncDataStore.pendingSyncCount()).to(equal(2))
+                    expect(syncDataStore.pendingSyncEntities().count).to(equal(2))
+                    expect(syncDataStore.pendingSyncOperations().count).to(equal(1))
+                
+                    let entities = kinveyFind(dataStore: syncDataStore).entities
+                    expect(entities?.count).to(equal(2))
+                }
+                it("should use multi-insert even if the items have not been created in an array") {
+                    var postCount = 0
+                    mockResponse { request in
+                        switch request.httpMethod {
+                        case "POST":
+                            postCount += 1
+                            fallthrough
+                        default:
+                            return HttpResponse(request: request, urlProcotolType: KinveyURLProtocol.self)
+                        }
+                    }
+                    defer {
+                        setURLProtocol(nil)
+                        expect(postCount).to(equal(1))
+                    }
+                    
+                    let entities = kinveySave(dataStore: syncDataStore, numberOfItems: 2).entities
+                    expect(entities?.count).to(equal(2))
+                    
+                    expect(syncDataStore.pendingSyncCount()).to(equal(2))
+                    expect(syncDataStore.pendingSyncEntities().count).to(equal(2))
+                    expect(syncDataStore.pendingSyncOperations().count).to(equal(1))
+                    
+                    let count = kinveyPush(dataStore: syncDataStore).count
+                    expect(count).to(equal(2))
+                    
+                    expect(syncDataStore.pendingSyncCount()).to(equal(0))
+                    expect(syncDataStore.pendingSyncEntities().count).to(equal(0))
+                }
+                it("Sync()") {
+                    mockResponse { request in
```

Comment:
Victor, It would be best if this test follows the correct flow disregarding the bug in KCS that returns a single error instead of returning success for the 2 correct items and an error for the invalid one. And just skip it until the issue is fixed. So the correct behavior would be for sync to return success for items 0 and 2 and error for item 1. So the pending item would only be item 1. Also, in the file I forgot to name the test. I have added a name: "should handle multi-insert push error properly"

Reply:
done

Reply:
This test should be the same as the last one - but for sync store